### PR TITLE
[CPU] Postpone throwing error for dnnl memory until it's requested

### PR DESCRIPTION
**Pull request by maxnick**
### Details:
StaticMemory object tries to create dnnl::memory object in the constructor. However, this memory object is used only by dnnl primitives and may be never requested if it isn't an input of a Node that uses a oneDNN primitive. However, it's still better to try to create the object inside the constructor to avoid data race issues. So the solution is to try to create dnnl::memory object and throw only when it's requested instead of throwing from the constructor.

### Tickets:
 - 121363
 - 120279

### CODE REVIEW

**Comment by maxnick** [comment]
@riverlijunjie , FIY

File: src/plugins/intel_cpu/src/cpu_memory.cpp [hunk 1]
```diff
@@ -288,24 +288,28 @@ StaticMemory::StaticMemory(const dnnl::engine& eng, MemoryDescPtr desc, const vo
...
+        //
+        // ========================
+        if (pads_zeroing)
+            m_prim.set_data_handle(m_pMemMngr->getRawPtr());
+        else
+            m_prim.set_data_handle_no_pads_proc(m_pMemMngr->getRawPtr());
+    }
+    catch (const std::exception& exc) {
+        dnnlErrorCtx = exc.what();
+    }
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/cpu_memory.cpp:312]: It does can handle "const->result" graph for unsupported precision, but it seems it cannot handle "const->convert->result" graph, such as:
`Constant(u16)->convert(u16->f32)->result(f32)`

Could you create such test cases?

File: src/plugins/intel_cpu/src/dnnl_extension_utils.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ memory::data_type DnnlExtensionUtils::IEPrecisionToDataType(const InferenceEngin
         case InferenceEngine::Precision::UNSPECIFIED:
             return memory::data_type::undef;
         default: {
-            OPENVINO_THROW("The plugin does not support ", prec.name());
+            IE_THROW() << "OneDNN doesn't support " << prec.name();
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/dnnl_extension_utils.cpp:71]: It will be better to restore this change.

Comment (maxnick) [reply src/plugins/intel_cpu/src/dnnl_extension_utils.cpp:71]: Yep, agree, we shouldn't use `IE_THROW()` as a deprecated macro.

File: src/plugins/intel_cpu/src/cpu_memory.cpp (hunk 1 above)
Comment (maxnick) [reply src/plugins/intel_cpu/src/cpu_memory.cpp:318]: Aren't the enabled tests enough? Your proposed subgraph will likely be folded during the constant folding pass.

**Comment by maxnick** [comment]
@dmitry-gorokhov , could you please take a look?

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**